src/connect/components/drive-view/drive-view.stories.tsx

```diff
@@ -40,132 +69,56 @@ const meta: Meta<typeof DriveView> = {
 export default meta;
 type Story = StoryObj<typeof meta>;
 
-const items = [
-    {
-        id: 'drive/folder1',
-        label: 'Folder 1',
-        type: ItemType.Folder,
-        status: ItemStatus.Syncing,
-        expanded: false,
-        children: [
-            {
-                id: 'drive/folder1/folder1.1',
-                label: 'Folder 1.1',
-                type: ItemType.Folder,
-                status: ItemStatus.Syncing,
-                expanded: false,
-            },
-            {
-                id: 'drive/folder1/folder1.2',
-                label: 'Folder 1.2',
-                type: ItemType.Folder,
-                status: ItemStatus.Syncing,
-                expanded: false,
-                children: [
-                    {
-                        id: 'drive/folder1/folder1.2/folder1.2.1',
-                        label: 'Folder 1.2.1',
-                        type: ItemType.Folder,
-                        status: ItemStatus.Syncing,
-                        expanded: false,
-                    },
-                ],
-            },
-        ],
-    },
-    {
-        id: 'drive/folder2',
-        label: 'Folder 2',
-        type: ItemType.Folder,
-        status: ItemStatus.AvailableOffline,
-        expanded: false,
-        children: [
-            {
-                id: 'drive/folder2/folder2.1',
-                label: 'Folder 2.1',
-                type: ItemType.Folder,
-                status: ItemStatus.AvailableOffline,
-                expanded: false,
-            },
-        ],
-    },
-    {
-        id: 'drive/folder3',
-        label: 'Folder 3',
-        type: ItemType.Folder,
-        status: ItemStatus.Offline,
-        expanded: false,
-    },
-];
-
 const DriveViewImpl = (args: DriveViewProps) => {
-    const { drives: initialDrives, onItemClick, ...restArgs } = args;
-    const [drives, setDrives] = useState(initialDrives);
+    const { onItemClick, ...restArgs } = args;
+    const { setItems } = useItemsContext();
+    const getDriveParent = useGetDriveParent();
 
     const onItemClickHandler: DriveViewProps['onItemClick'] = (
         e,
         item,
-        drive,
     ) => {
-        const newDrives = traverseDriveById(drives, drive.id, treeItem => {
-            if (treeItem.id === item.id) {
+        setItems((prevItems) => prevItems.map((prevItem) => {
+            if (prevItem.id === item.id) {
                 return {
-                    ...treeItem,
-                    expanded: !treeItem.expanded,
+                    ...prevItem,
+                    isSelected: true,
+                    expanded: !prevItem.expanded,
                 };
             }
 
-            return treeItem;
-        });
+            return { ...prevItem, isSelected: false };
+        }));
 
-        setDrives(newDrives);
-        onItemClick?.(e, item, drive);
+        const parent = getDriveParent(item.path);
```

Comment:
This is how you can get the drive of any item (no need to pass the drive in the callback functions)